megatron/core/inference/contexts/dynamic_context.py

```diff
@@ -384,36 +390,25 @@ def __init__(
 
         # Initialize block allocator.
         buffer_size_bytes = int(buffer_size_gb * 1024**3)
-        block_count_total = buffer_size_bytes // (
+        paused_buffer_size_bytes = (
+            0
+            if paused_buffer_size_gb is None else
+            int(paused_buffer_size_gb * 1024**3)
+        )
+        block_count = buffer_size_bytes // (
+            self.block_size_bytes + mamba_states_memory_per_request
+        )
```

Comment:
Can you add a TODO here to revisit making this allocation decision more flexible with respect to how much memory we want to give to KV cache versus Mamba state? I believe vLLM and/or SGLang have a parameter that lets you control the fraction assigned to each.